rust/kernel/spi.rs

```diff
@@ -0,0 +1,296 @@
+// SPDX-License-Identifier: GPL-2.0
+
+//! This module provides safer and higher level abstraction over the kernel's SPI types
+//! and functions.
+//!
+//! C header: [`include/linux/spi/spi.h`](../../../../include/linux/spi/spi.h)
+
+use crate::bindings;
+use crate::c_types;
+use crate::error::{code::*, Error, Result};
+use crate::str::CStr;
+use alloc::boxed::Box;
+use core::pin::Pin;
+
+/// Wrapper struct around the kernel's `spi_device`.
+#[derive(Clone, Copy)]
```

Comment:
Wouldn't this allow me to clone a valid SpiDevice only to have it being deleted eventually while I am still holding a SpiDevice to a dangling pointer which would result in unsafe behaviour when calling the safe `write` functions below?

Reply:
Yes I think that's a problem. I will try to find some time to fix this PR as Rust for Linux has evolved quite a lot and I'm not sure if this is still even compiling